lab_2_retrieval_w_bm25/main.py

```diff
@@ -77,8 +132,25 @@ def calculate_idf(vocab: list[str], documents: list[list[str]]) -> dict[str, flo
 
     In case of corrupt input arguments, None is returned.
     """
-
-
+    if (not isinstance(vocab, list) or not isinstance(documents, list)
+            or len(vocab) == 0 or len(documents) == 0):
+        return None
+    idf = {}
+    for elem in vocab:
+        if not isinstance(elem, str):
+            return None
+        doc_number = 0
+        for doc in documents:
+            if not isinstance(doc, list):
+                return None
+            for word in doc:
+                if not isinstance(word, str):
+                    return None
+            if elem in doc:
+                doc_number += 1
+                continue
```

Comment:
is it really need to use `continue` here

Reply:
Fixed